lib/Analysis/MulDepthAnalysis/MulDepthAnalysis.h

```diff
@@ -9,31 +9,75 @@
 namespace mlir {
 namespace heir {
 
+/// This class represents the multiplicative depth in an FHE program.
+///
+/// The multiplicative depth in an FHE program refers to the maximum number of
+/// multiplicative operations that can be performed on encrypted data before the
+/// noise introduced during encryption renders the data undecryptable.
+///
+/// Cryptographic parameters, especially the `Coefficient Modulus`, directly
+/// influence the amount of noise that accumulates during these operations.
+/// Therefore, when determining the cryptographic parameters, it is essential to
+/// consider the required multiplicative depth to ensure that the noise level
+/// remains manageable and the encrypted data can be correctly decrypted after
+/// the necessary computations.
+///
+/// It tracks the depth of nested multiplications, accounting for
+/// non-multiplicative operations in between. For example, the multiplicative
+/// depth of the following pseudo-code is 2:
+///
+///     %0 = arith.constant 0 : multiplicative depth = 0
+///     %1 = FHE.mul %0 %0 : multiplicative depth = 1
+///     %2 = FHE.add %0 %1 : multiplicative depth = 1
+///     %3 = FHE.mul %2 %0 : multiplicative depth = 2
+///
+/// Formally speaking, the multiplicative depth for an arbitrary operation `op`
+/// can be defined inductively as follows.
+///
+///     \[
+///     \text{MulDepth}(\text{op}) =
+///     \begin{cases}
+///     \max\limits_{i}
+///     \left(\text{MulDepth}(\text{operand}_i(\text{op}))\right) + 1 & \text{if
+///     op is multiplication} \\ \max\limits_{i}
+///     \left(\text{MulDepth}(\text{operand}_i(\text{op}))\right) &
+///     \text{otherwise} \end{cases}
+///     \]
+
```

Comment:
I'm not sure this latex comment can be supported?

Reply:
It is not supported in the comments here, but if you put it in the tablegen, it will be rendered as latex on heir.dev

Reply:
What about this pseudocode instead?

```
MulDepth(z) = case
  mul_op(x, y):
    max(MulDepth(x), MulDepth(y)) + 1
  any_op(operands):
    max(map(MulDepth, operands))
```

Reply:
nice!